### ISSUE TITLE
Android background data messages not being delivered to app

### ISSUE THREAD

**Issue by richmartin** [phonegap/phonegap-plugin-push#942]
### Expected Behaviour
When the app is in the background and a data-only message is received from Google Cloud Messaging with content_available set to true, the app should receive the message.

### Actual Behaviour
The message is  received by GCMIntentService, but not delivered to the app.

### Reproduction Steps
Send a message to the device with the following structure:
```
    {
      to: '<topic or device id>',
      priority: 'normal',
      content_available: true,
      delay_while_idle: false,
      data: {
        some_app_specific_data: 'something'
      }
    }
```
while monitoring the Android log of the device.

The log shows the message being received by GCMIntentService but not delivered to the app.

Debugging shows that the GCMIntentService is looking for a message property `content-available` equal to the string "1", rather than the correct `content_available` equal to `true`.

The content_available option is documented here: https://developers.google.com/cloud-messaging/http-server-ref#downstream-http-messages-json

Note that the correct behaviour (according to the documentation) would be to _always_ deliver data messages regardless of the value of the content_available attribute; i.e. to remove the `if ("true".equals(contentAvailable))` conditional around `PushPlugin.sendExtras(extras)` on line 216 of GCMIntentServer.java.

> On iOS, use this field to represent content-available in the APNS payload. When a notification or message is sent and this is set to true, an inactive client app is awoken. On Android, data messages wake the app by default. On Chrome, currently not supported.

 Alternatively, if the desire to respect the value of `content_available` on Android (which would be a deviation from the documented behaviour of this option), then the value of `CONTENT_AVAILABLE` in `PushConstants.java` should be changed from `content-available` to `content_available` and the conditional in `GCMIntentServer` should be changed from `if ("1".equals(contentAvailable))` to `if ("true".equals(contentAvailable))`.

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
Android, all versions.

**Comment by macdonst** [comment]
@richmartin Yeah, I have to play a bit fast and loose with the "correct behaviour". Originally I tried to line up with what iOS was doing but you make a good point. I'm going to leave the original code in there but also look for "content_available: true" in the payload and treat it the same way.

**Comment by macdonst** [comment]
@richmartin Yeah, I don't think I can change this. Possibly the way FCM is implemented by Google may change things.

**Comment by richmartin** [comment]
Are you sure? I'm running with a modified version that looks for `content_available` equal to `"true"` and they are coming through fine (on Android 6.0.1). I see the key being changed by the key normalising method (stripping it of the prefix).

Also, I strongly believe that the correct behaviour is to ignore this parameter and to to always deliver background data messages. That is the documented GCM behaviour and what users of the GCM API will expect, and it's consistent with the behaviour of native apps.

Can you think of a situation where silently dropping the message is better than delivering it and letting the app decide how to deal with it?

**Comment by macdonst** [comment]
@richmartin well if you can do a PR with your modified version I'd be grateful.

The reason I don't always call the `notification` event when a push with the `data` payload is received is I'm trying to give developers more control over when the event will be called. Because of the way the `notification` payload is handled, the OS takes over and we never get an event, I need to use the `data` payload for everything. If the `content-available` property is set to `1` then I call the `notification` event when the app is in the background, if it is not set the `notification` event is not called unless the user clicks on the notification in the notification shade.

I would love to move completely to the way things work in GCM but I have to balance it with supporting calling JS methods when action buttons are clicked and the whole notification payload problem.
